src/NexusMods.App.UI/Controls/TreeDataGrid/SharedColumns.cs

```diff
@@ -57,4 +57,23 @@ public static int Compare<TKey>(CompositeItemModel<TKey> a, CompositeItemModel<T
         public static string GetColumnHeader() => "Size";
         public static string GetColumnTemplateResourceKey() => ColumnTemplateResourceKey;
     }
+
+    [UsedImplicitly]
+    public sealed class Value<TValue> : ICompositeColumnDefinition<Value<TValue>>
+    {
+        public static int Compare<TKey>(CompositeItemModel<TKey> a, CompositeItemModel<TKey> b) where TKey : notnull
+        {
+            var aValue = a.GetOptional<ValueComponent<TValue>>(ComponentKey);
+            var bValue = b.GetOptional<ValueComponent<TValue>>(ComponentKey);
+            // Assuming ValueComponent has a comparable Value property or implements IComparable
+            // Adjust the comparison logic if ValueComponent comparison needs specific handling
+            return aValue.Compare(bValue); 
+        }
+
+        public const string ColumnTemplateResourceKey = Prefix + nameof(Value<TValue>);
+        public static readonly ComponentKey ComponentKey = ComponentKey.From(ColumnTemplateResourceKey + "_" + nameof(ValueComponent<TValue>));
+
+        public static string GetColumnHeader() => "Value";
+        public static string GetColumnTemplateResourceKey() => ColumnTemplateResourceKey;
+    }
```

Comment:
Columns should be concrete. If you need a column to show the number of files in a directory then you should make that column concrete.